rclcpp_lifecycle/include/rcl_lifecycle/transition_map.h

```diff
@@ -0,0 +1,101 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef RCL__TRANSITION_MAP_H_
+#define RCL__TRANSITION_MAP_H_
+
+#include <rcl_lifecycle/lifecycle_state.h>
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+typedef struct _rcl_state_t rcl_state_t;
+
+typedef struct _rcl_state_transition_t rcl_state_transition_t;
+
+/**
+ * @brief The actual data array
+ * within the map. One array belongs
+ * to one label within the map.
+ */
+typedef struct LIFECYCLE_EXPORT _data_array
+{
+  rcl_state_transition_t* transitions;
+  unsigned int size;
+} rcl_transition_array_t;
+
+/**
+ * @brief helper index for keeping
+ * track of the map content
+ */
+typedef struct LIFECYCLE_EXPORT _index
+{
+  unsigned int index;
+  const char* label;
```

Comment:
`label` here was meant to be a human readable string such as "active" or "inactive".